what4/src/What4/Expr/App.hs

```diff
@@ -498,6 +494,12 @@ data App (e :: BaseType -> Type) (tp :: BaseType) where
     -> App e (BaseBVType w)
   FloatToReal :: !(e (BaseFloatType fpp)) -> App e BaseRealType
 
+  FloatSpecialFunction ::
```

Comment:
Out of curiosity, would it make sense to have a similar `SpecialFunction` construct for complex numbers (e.g., to support the operations in `complex.h`)?

Reply:
Good question, I haven't thought too much about how to represent `complex.h`.  It seems to me like the intended implementation there is that `complex` is just a pair of reals in rectangular coordinates, so I'd probably do that instead of directly representing complex floating-point values.

Reply:
Indeed, it seems that clang boils away the handling of complex values into corresponding functions on reals.